UnicornRender/video/source/Material.cpp

```diff
@@ -104,5 +105,62 @@ glm::vec3 Material::GetColor() const
     return m_color;
 }
 
+void Material::SetTextureCoordinates(int32_t x, int32_t y, int32_t width, int32_t height)
+{
+    if (!m_albedo->IsLoaded())
+    {
+        LOG_WARNING("Setting sprite coordinates, when texture was not set, nothing will be set.");
+        return;
+    }
+
+    m_spriteCoordinates = { static_cast<float>(x) / m_albedo->Width(),
+        static_cast<float>(y) / m_albedo->Height(),
+        static_cast<float>(width) / m_albedo->Width(),
+        static_cast<float>(height) / m_albedo->Height() };
+
+    DataUpdated.emit();
+}
+
+void Material::SetNormalizedTextureCoordinates(float x, float y, float width, float height)
+{
+    if (!m_albedo->IsLoaded())
+    {
+        LOG_WARNING("Setting sprite coordinates, when texture was not set, nothing will be set.");
+        return;
+    }
+
+    m_spriteCoordinates = { x, y, width, height };
+
+    DataUpdated.emit();
+}
+
+
+void Material::SetDefaultTextureCoodinates()
+{
+    m_spriteCoordinates = { 0, 0, 1.0, 1.0 };
```

Comment:
I feel like this method is not needed - `SetNormalizedTextureCoordinates()` can be used instead